apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/CallbackAdapterInterceptor.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.kafka;
+
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+
+import java.lang.reflect.Method;
+
+/**
+ * implements Callback and EnhancedInstance, for kafka callback in lambda expression
+ */
+public class CallbackAdapterInterceptor implements Callback, EnhancedInstance {
+
+    private Method method;
+
+    private CallbackCache callbackCache;
+
+    private static CallbackInterceptor CALLBACK_INTERCEPTOR = new CallbackInterceptor();
+
+    private ILog logger = LogManager.getLogger(CallbackAdapterInterceptor.class);
+
+    public CallbackAdapterInterceptor(CallbackCache callbackCache) {
+
+        this.callbackCache = callbackCache;
+        try {
+            this.method = CallbackAdapterInterceptor.class.getMethod("onCompletion", new Class[]{RecordMetadata.class, Exception.class});
+        } catch (Exception e) {
+        }
+    }
+
+    @Override
+    public void onCompletion(RecordMetadata metadata, Exception exception) {
```

Comment:
In this method, why so many try-catch? Super confusing about what you are doing. Could you explain?

Reply:
This class should just make sure the original callback being traced, but currently, I can't follow your point.